src/server/game/Spells/Spell.cpp

```diff
@@ -7129,13 +7129,7 @@ void Spell::DoAllEffectOnLaunchTarget(TargetInfo& targetInfo, float* multiplier)
         }
     }
 
-    if (Player* modOwner = m_caster->GetSpellModOwner())
-        modOwner->SetSpellModTakingSpell(this, true);
```

Comment:
If you remove this, SPELLMOD_CRIT won't register mod in auras, preventing charges dropping. If this is intended then 4829cca9fb81b27cd9a057355ee7b03ff71e0a14 shall be reverted